src/utils.jl

```diff
@@ -1,3 +1,26 @@
+#-----------------------------------------------------------------------# Missing/DataValue
+missing_instance(::Type{Missing}) = missing 
+missing_instance(::Type{DataValue}) = DataValue()
+
+type2missingtype(T, ::Type{Missing}) = Union{T, Missing} 
+type2missingtype(T, ::Type{DataValue}) = DataValue{T}
+type2missingtype(T::Type{<:DataValue}, ::Type{DataValue}) = T
+
+missingtype2type(T) = Base.nonmissingtype(T)
+missingtype2type(::Type{DataValue{T}}) where {T} = T
+
+unwrap(x) = x
+unwrap(x::DataValue) = get(x)
+
+ismissingtype(T, ::Type{Missing}) = Missing <: T 
+ismissingtype(T, ::Type{DataValue}) = T <: DataValue
+
+_ismissing(x) = ismissing(x)
+_ismissing(x::DataValue) = isna(x)
+
+Base.similar(T::Type{DataValueArray{S}}, n::Int) where {S} = DataValueArray(Vector{S}(undef, n))
```

Comment:
Can we dispatch on `n::Dims` instead? Or potentially add two methods? The machinery in StructArrays needs the `similar(T, ::Dims)` method.

Reply:
done.